portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -0,0 +1,93 @@
+package com.google.sps.servlets;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.sps.Constants;
+
+/** Servlet that processes comment. */
+@WebServlet("/comment")
+public final class CommentsServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.        
+		BufferedReader br = 
+		new BufferedReader(new InputStreamReader(request.getInputStream()));
+		String newComment = "";
+		if(br != null){
+			newComment = br.readLine();
+			System.out.println(newComment);
+		}
+
+    //convert to object
+    Gson g = new Gson(); 
+    Comment c = g.fromJson(newComment, Comment.class);
+
+    //create comment entity
+    Entity commentEntity = new Entity(Constants._COMMENT);
+    commentEntity.setProperty(Constants.NAME, c.getName());
+    commentEntity.setProperty(Constants.COMMENT,c.getComment());
+    
+
+    //Store comment in datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
It would be better to initialize the service, at the time of class initialization, like in a constructor. Instead of fetching it everytime.